coordinator/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/StargateQueryHandler.java

```diff
@@ -130,8 +130,11 @@ public Single<ResultMessage.Prepared> prepare(
         .prepare(query, queryState, customPayload)
         .map(
             p -> {
-              Prepared prepared = QueryProcessor.instance.getPrepared(p.statementId);
-              CQLStatement statement = prepared.statement;
+              // we need to protect against the case where the statement was dropped from the cache
+              CQLStatement statement =
+                  Optional.ofNullable(QueryProcessor.instance.getPrepared(p.statementId))
+                      .map(prepared -> prepared.statement)
+                      .orElseGet(() -> QueryProcessor.getStatement(query, queryState));
```

Comment:
Statement object itself needs to be somewhat valid for `IdempotencyAnalyzer`.

...oh, except that it uses `instanceof` for everything so I guess `null` would pass through and just return `true.
So that wouldn't require valid instance after all.